packages/web5/test/pexv2/pd_test.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:test/test.dart';
+import 'package:web5/src/pexv2/pd.dart';
+
+import '../helpers/test_vector_helpers.dart';
+
+
+class SelectCredentialTestVector {
+  String description;
+
+  // Input
+  PresentationDefinition inputPresentationDefinition;
+  List<String> inputVcJwts;
+
+  // output
+  List<String> outputSelectedCredentials;
+  bool errors;
+
+  SelectCredentialTestVector({
+    required this.description,
+    required this.inputPresentationDefinition,
+    required this.inputVcJwts,
+    required this.outputSelectedCredentials,
+    this.errors = false,
+  });
+
+  factory SelectCredentialTestVector.fromJson(Map<String, dynamic> json) {
+    return SelectCredentialTestVector(
+      description: json['description'],
+      inputPresentationDefinition: PresentationDefinition.fromJson(
+        json['input']['presentationDefinition'],
+      ),
+      inputVcJwts: json['input']['credentialJwts'],
+      outputSelectedCredentials: json['output']['selectedCredentials'],
+      errors: json['errors'],
+    );
+  }
+}
+
+void main() {
+  group('select credentials', () {
+    group('vectors', () {
+      late List<SelectCredentialTestVector> vectors;
+
+      setUpAll(() {
+        final vectorsJson = getJsonVectors('presentation_exchange/select_credentials.json');
```

Comment:
running into trouble reading the vectors from the submodule. i'll take another stab at it in the morning